homework/index.js

```diff
@@ -30,15 +26,137 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  const root = document.getElementById('root');
+  const container = createAndAppend('div', root, { id: 'container' });
+
+  function getSelectedData(value, data) {
+    let selected;
+    data.forEach(item => {
+      if (item.name === value) {
+        selected = item;
       }
     });
+    return selected;
+  }
+
+  function generateSelections(repos) {
+    const header = createAndAppend('header', root, {});
+    const selectionElem = createAndAppend('select', header, {
+      id: 'repositories',
+    });
+
+    repos.forEach(repo => {
+      createAndAppend('option', selectionElem, {
+        id: repo.id++,
```

Comment:
You are recieving a ready `id` making the item identified without any kind of repetition, here you have two main things to have an attention:
  1. you are changing the original id, for future when you recieve a one don't change it on the client, if you don't have you may create a one, but not when you have it already.
  
  2. you are creating `id`s attribute for each option you have recieved as a repo name, means if you have 100 option then you have 100 different `id` you will never use, here or in your Css.